### PR TITLE
Introduce a common function to retrieve the `eps` quantity

### PR DESCRIPTION
**Pull request by elehcim**
Tidy up `eps` usage.
I added the function `get_eps` in util.py which tries to retrieve `eps` taking care of the correct dtype which sometimes could create problems (e.g. dtype mismatch like the one in #355).

### CODE REVIEW

**Comment by apontzen** [comment]
Thanks, this looks great as soon if it can be made compatible with python <3.5 (unclear to me why the travis tests are failing for those python versions at a glance)

File: pynbody/gravity/calc.py [hunk 1]
```diff
@@ -83,14 +83,7 @@ def midplane_rot_curve(f, rxy_points, eps=None, mode=config['gravity_calculation
             "OpenMP module is now selected at install time", DeprecationWarning)
 
     if eps is None:
-        try:
-            eps = f['eps']
-        except KeyError:
-            eps = np.zeros(len(f))
-            eps += f.properties['eps']
-
-    if isinstance(eps, str):
```
Comment (apontzen) [inline pynbody/gravity/calc.py]: For backwards compatibility I think we need to allow eps to be a string that gets auto-converted into a unit. This doesn't seem to be possible in the new version.

Comment (elehcim) [reply pynbody/gravity/calc.py]: The prerequisite of the function `direct` in _gravity.pyx is that eps should be a SimArray of the same dtype as `f['mass']` and the same units as `f['pos']`. For that before every call to `direct` I think it is good to make sure `eps` is a SimArray.

**Comment by apontzen** [comment]
Many thanks for this excellent patch!

**Pull request merged**